src/charm.py

```diff
@@ -52,11 +56,17 @@ def __init__(self, *args):
             self._on_nginx_pebble_ready,
         )
 
+        self._s3_storage_data = _S3StorageBackend()
```

Comment:
the identifier name should match the name of the type:
either `_s3_storage_backend = ...` or `... = _S3StorageData`